lib/sqlalchemy/sql/elements.py

```diff
@@ -2094,7 +2118,22 @@
             )
         return c
 
-    def _gen_cache_key(self, anon_map, bindparams):
+    def _gen_cache_key(
+        self,
+        anon_map: cache_anon_map,
+        bindparams: List[BindParameter[_T]],
+    ) -> Optional[
+        Union[
```

Comment:
**mike bayer** (zzzeek) wrote:

this whole thing I think overall is just `Tuple[Any]`, it wasn't intended to be specific to what's in the tuple.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4512